src/Controller/Component/AuthComponent.php

```diff
@@ -817,7 +821,7 @@ public function constructAuthenticate()
                 $class = $alias;
             }
             $className = App::className($class, 'Auth', 'Authenticate');
-            if (!class_exists($className)) {
+            if ($className === false || !class_exists($className)) {
                 throw new Exception(sprintf('Authentication adapter "%s" was not found.', $class));
             }
             if (!method_exists($className, 'authenticate')) {
```

Comment:
Do we need to keep this check in 4.x+? Maybe we can also use some interface approach in the future.

Reply:
AuthComponent and related classes themselves will go away in a future release, so not worth the effort.

Reply:
@ADmad Done, let me know if it's ok this way :)